src/model/HeatPumpPlantLoopEIRHeating.cpp

```diff
@@ -0,0 +1,491 @@
+      : WaterToWaterComponent_Impl(idfObject, model, keepHandle) {
+      OS_ASSERT(idfObject.iddObject().type() == HeatPumpPlantLoopEIRHeating::iddObjectType());
+    }
+
+    HeatPumpPlantLoopEIRHeating_Impl::HeatPumpPlantLoopEIRHeating_Impl(const openstudio::detail::WorkspaceObject_Impl& other, Model_Impl* model,
+                                                                       bool keepHandle)
+      : WaterToWaterComponent_Impl(other, model, keepHandle) {
+      OS_ASSERT(other.iddObject().type() == HeatPumpPlantLoopEIRHeating::iddObjectType());
+    }
+
+    HeatPumpPlantLoopEIRHeating_Impl::HeatPumpPlantLoopEIRHeating_Impl(const HeatPumpPlantLoopEIRHeating_Impl& other, Model_Impl* model,
+                                                                       bool keepHandle)
+      : WaterToWaterComponent_Impl(other, model, keepHandle) {}
+
+    const std::vector<std::string>& HeatPumpPlantLoopEIRHeating_Impl::outputVariableNames() const {
+      static const std::vector<std::string> result{"Heat Pump Electricity Energy",
+                                                   "Heat Pump Load Side Heat Transfer Energy",
+                                                   "Heat Pump Source Side Heat Transfer Energy",
+                                                   "Heat Pump Electricity Rate",
+                                                   "Heat Pump Load Side Heat Transfer Rate",
+                                                   "Heat Pump Source Side Heat Transfer Rate",
+                                                   "Heat Pump Load Side Outlet Temperature",
+                                                   "Heat Pump Load Side Inlet Temperature",
+                                                   "Heat Pump Source Side Outlet Temperature",
+                                                   "Heat Pump Source Side Inlet Temperature",
+                                                   "Heat Pump Load Side Mass Flow Rate",
+                                                   "Heat Pump Source Side Mass Flow Rate"};
+      return result;
+    }
+
+    IddObjectType HeatPumpPlantLoopEIRHeating_Impl::iddObjectType() const {
+      return HeatPumpPlantLoopEIRHeating::iddObjectType();
+    }
+
+    unsigned HeatPumpPlantLoopEIRHeating_Impl::supplyInletPort() const {
+      return OS_HeatPump_PlantLoop_EIR_HeatingFields::SourceSideInletNodeName;
+    }
+
+    unsigned HeatPumpPlantLoopEIRHeating_Impl::supplyOutletPort() const {
+      return OS_HeatPump_PlantLoop_EIR_HeatingFields::SourceSideOutletNodeName;
+    }
+
+    unsigned HeatPumpPlantLoopEIRHeating_Impl::demandInletPort() const {
+      return OS_HeatPump_PlantLoop_EIR_HeatingFields::LoadSideInletNodeName;
+    }
+
+    unsigned HeatPumpPlantLoopEIRHeating_Impl::demandOutletPort() const {
+      return OS_HeatPump_PlantLoop_EIR_HeatingFields::LoadSideOutletNodeName;
+    }
```

Comment:
Same, I think this is backwards.

Reply:
I'm not going to bother re-adding all comments. All comments that apply to the Cooling one also apply to the Heating one